controllers/database_controller_test.go

```diff
@@ -25,6 +25,8 @@ import (
 	"context"
 	"time"
 
+	b64 "encoding/base64"
```

Comment:
Would it be possible to cover more negative test cases. For instance, can we check the negative test scenario that all the errors are caught?

Reply:
Since it is not straightforward to have a functioning mock infrastructure (mocked NDB server + mocked Kubernetes cluster + communication between these mocks) it is not a feasible option to test this controller/reconcile logic (cannot inject mocks for various cases). However, I have added negative test cases for the pieces of codes / helper methods that this controller calls for the expected error cases.